### PR TITLE
fix(Scripts/Pets): Darting Hatchling not running around.

### PR DESCRIPTION
**Pull request by avarishd**
## Changes Proposed:
- Change from a generic pet to a scripted creature.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/12790
- Closes https://github.com/chromiecraft/chromiecraft/issues/3951

## Tests Performed:

- tested

## How to Test the Changes:

.learn 67413
Confirm that every 20s, runs for about 2s and then comes back.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Maelthyrr** [comment]
Hi, thanks for the PR. Tested and work fine.

File: src/server/scripts/Pet/pet_generic.cpp [hunk 1]
```diff
@@ -774,6 +774,50 @@ struct npc_pet_gen_moth : public NullCreatureAI
     }
 };
 
+// Darting Hatchling
+enum Darting
+{
+    SPELL_DARTING_ON_SPAWN      = 62586, // Applied on spawn via creature_template_addon
+    SPELL_DARTING_FEAR          = 62585, // Applied every 20s from SPELL_DARTING_ON_SPAWN
+
```
Comment (IntelligentQuantum) [inline src/server/scripts/Pet/pet_generic.cpp:782]: remove empty line :fearful:

**Review by IntelligentQuantum — APPROVED** [review]

**Comment by Nyeriah** [comment]
Conflicts here

File: src/server/scripts/Pet/pet_generic.cpp [hunk 2]
```diff
@@ -774,6 +774,49 @@ struct npc_pet_gen_moth : public NullCreatureAI
...
+    {
+        if (!goFast)
+        {
+            return;
+        }
+
+        checkTimer += diff;
+        if (checkTimer >= 2000)
+        {
+            me->RemoveAura(SPELL_DARTING_FEAR);
```
Comment (Kitzunu) [inline src/server/scripts/Pet/pet_generic.cpp:813]:
```suggestion
            me->RemoveAuraDueToSpell(SPELL_DARTING_FEAR);
```

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: src/server/scripts/Pet/pet_generic.cpp (hunk 2 above)
Comment (avarishd) [reply src/server/scripts/Pet/pet_generic.cpp:813]: Missed and "S" there. 😇

Comment (Kitzunu) [reply src/server/scripts/Pet/pet_generic.cpp]: The reason for the change is that RemoveAura removes the Aura, whilist RemoveAuraDueToSpells will remove the spell

**Pull request merged**